packages/calcite-components/src/components/input-date-picker/input-date-picker.tsx

```diff
@@ -158,21 +158,31 @@ export class InputDatePicker
   @Watch("value")
   valueWatcher(newValue: string | string[]): void {
     if (!this.userChangedValue) {
-      let newValueAsDate;
-      if (Array.isArray(newValue)) {
+      const valueIsArray = Array.isArray(newValue);
+      let newValueAsDate: Date | Date[];
+
+      if (valueIsArray) {
         newValueAsDate = getValueAsDateRange(newValue);
       } else if (newValue) {
         newValueAsDate = dateFromISO(newValue);
       } else {
         newValueAsDate = undefined;
       }
 
+      const { focusedInput } = this;
+
+      this.setInputValue(
+        valueIsArray ? newValue[focusedInput === "end" ? 1 : 0] : newValue,
+        focusedInput
+      );
+
```

Comment:
This looks good, the only thing I'm wondering is if we should move this logic over to the `localizeInputValues()` method?  I'm realizing that method only handles localizing truthy input values, but it could just as well handle empty values as well.  Up to you, no hard opinion either way.